## Домашнее задание №1
***

### Задача 1
Из колоды в 52 карты извлекаются случайным образом 4 карты \
  a) найти вероятность того, что все карты – крести;\
  б) найти вероятность, что среди 4-х карт окажется хотя бы один туз.

#### Решение
для определения вероятности воспользуемся формулой классической вероятности $ P(A)=\frac{m}{n} $,\
где ***m*** - число благоприятных исходов, ***n*** - общее количество исходов.

Для вычисления количества исходов, используем формулу количества сочетаний $ C\binom{n}{k} = \frac{n!}{k!(n-k)!} $,\
где ***n*** - количество элементов (множество), ***k*** - количество сочетаний.

Для расчета будем использовать функцию *get_combinations*, которая на вход принимает ***n*** и ***k***, возвращает количество сочетаний.

а) вычислим ***m*** (в колоде 13 карт крестовой масти (52/4), из колоды вынимается 4 карты)

In [1]:
from math import factorial

def get_combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k))) 

In [2]:
m = get_combinations(13, 4)
print(f'm = {m}')

m = 715


вычислим ***n*** (общее количество карт в колоде - 52, вынимаем 4 карты)

In [3]:
n = get_combinations(52, 4)
print(f'n = {n}')

n = 270725


Определим вероятность того, что все карты - крести Р(4 карты - крести) 

In [4]:
P = m/n
print(f'Р(4 карты - крести) = {round(P, 4)}')

Р(4 карты - крести) = 0.0026


б) Общее количество исходов остается прежним n = 270725.\
Для определения количества благоприятных исходов (m), разобьем множество карт на подмножество тузов 4, и подмножество без тузов 48 (52-4).\
Количество благоприятных исходов будет складываться из случаев со следующими вариантами:
* из 4 карт 1 туз
* из 4 карт 2 туза
* из 4 карт 3 туза
* из 4 карт 4 туза (количество комбинаций 1)

In [5]:
m = get_combinations(48, 3)*get_combinations(4, 1) + get_combinations(48, 2)*get_combinations(4, 2) + get_combinations(48, 1)*get_combinations(4, 3) + 1
print(f'm = {m}')

m = 76145


In [6]:
P = m / n
print(f'P(хотя бы 1 туз) = {round(P, 2)}')

P(хотя бы 1 туз) = 0.28


так же эту задачу можно решить обратным способом:\
сумма вероятностей протовиположных событий $\displaystyle P(A)+P(\bar{A})=1$,\
противоположное событие - извлечь 4 карты без туза

In [7]:
m = get_combinations(48, 4)
print(f'm = {m}')

m = 194580


In [8]:
P = 1 - m/n
print(f'P(хотя бы 1 туз) = {round(P, 2)}')

P(хотя бы 1 туз) = 0.28


***
### Ответ:
**а) ***Р*** (4 карты - крести) ~ 0,0026**\
**б) ***P*** (хотя бы 1 туз) ~ 0,28**
***

### Задача 2
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

#### Решение
Для решения задачи будем использовать формулу количества сочетаний (т.к. порядок нажатия кнопок неважен, по условию задачи они нажимаются одновременно). Количество благоприятных исходов будет ***m = 1***, т.к. дверь должна быть открыта с первой попытки.

In [9]:
n =  get_combinations(10, 3)
P = 1 / n
print(f'P(открыть дверь с первой попытки) = {round(P, 4)}')

P(открыть дверь с первой попытки) = 0.0083


***
### Ответ:
*****Р*** (открыть дверь с первой попытки) ~ 0,0083**
***

### Задача 3
В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

#### Решение
Для решения задачи будем использовать формулу количества сочетаний (т.к. порядок извлечения деталей неважен).\
Определим общее количество исходов ***n*** из 15 деталей по 3.
Количество благоприятных исходов ***m*** определим из 9 окрашенных деталей по 3.

In [10]:
n =  get_combinations(15, 3)
print(f'n = {n}')

n = 455


In [11]:
m = get_combinations(9, 3)
print(f'm = {m}')

m = 84


In [14]:
P = m / n
print(f'Р(3 детали окрашены) = {round(P, 2)}')

Р(3 детали окрашены) = 0.18


***
### Ответ:
*****Р*** (3 детали окрашены) ~ 0,18**
***

#### Задача 4
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

#### Решение
Для решения задачи будем использовать формулу количества сочетаний.\
Определим общее количество исходов ***n*** из 100 билетов по 2.
Количество благоприятных исходов ***m*** = 1.

In [16]:
n =  get_combinations(100, 2)
print(f'n = {n}')

n = 4950


In [17]:
P=1/n
print(f'Р(2 билета выигрышные) = {round(P, 4)} ')

Р(2 билета выигрышные) = 0.0002 


***
### Ответ:
*****Р*** (2 билета выигрышные) ~ 0,0002**
***